### PR TITLE
Load plugins via sandbox separately from hooks

### PR DESCRIPTION
**Pull request by georgolden**
- [x] tests and linter show no problems (`npm t`)
- [x] tests are added/updated for bug fixes and new features
- [x] code is properly formatted (`npm run fmt`)

### CODE REVIEW

**Comment by georgolden** [comment]
@tshemsedinov cherry-pick this commit b75b28a642e2cd07b4afb7a1235d526ab4b3cf6c

**Comment by georgolden** [comment]
@tshemsedinov please cherry-pick this 529d124e3c68605725e4edbbb8ef96871077439e

File: test/api/geo.1.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+({
+  plugin: 'oldendevTestplugin/crud',
+  entities: { City: { name: 'string ' } },
```
Comment (tshemsedinov) [inline test/api/geo.1.js:3]:
```suggestion
  entities: { City: { name: 'string' } },
```

File: lib/interfaces.js [hunk 2]
```diff
@@ -79,20 +79,30 @@ class Interfaces extends Cache {
...
+    if (!lib || !lib.plugins) return;
+    const pluginSrc = lib.plugins[name];
+    if (!pluginSrc) return;
+    const context = this.application.sandbox;
+    const { exports } = metavm.createScript(name, pluginSrc, { context });
+    const plugin = exports(iface);
+    for (const [name, script] of Object.entries(plugin)) {
+      const proc = new Procedure(script, name, this.application);
+      this.changeInterface(interfaceName, name, proc);
+    }
```
Comment (tshemsedinov) [inline lib/interfaces.js:103]: Why do we need this code duplication with `.change` method ?

File: package.json [hunk 3]
```diff
@@ -67,6 +67,7 @@
   },
   "dependencies": {
     "metacom": "^3.0.0-alpha.4",
+    "@oldendev/testplugin": "^1.1.6",
```
Comment (tshemsedinov) [inline package.json:70]: Remove this

File: lib/dependencies.js [hunk 4]
```diff
@@ -34,6 +34,12 @@ const assignNamespace = (container, name, value) => {
   container[camelKey] = value;
 };
 
+const loadPlugins = (lib) => {
+  for (const plugin of Object.keys(lib.plugins)) {
+    lib.plugins[plugin] = String(lib.plugins[plugin]);
```
Comment (tshemsedinov) [inline lib/dependencies.js:39]:
```suggestion
    lib.plugins[plugin] = lib.plugins[plugin].toString();
```

File: test/interfaces.js [hunk 5]
```diff
@@ -14,11 +14,27 @@ const application = {
...
   },
+  Error,
 };
 
 metatests.testAsync('lib/interfaces load', async (test) => {
   const api = new Interfaces('api', application);
   await api.load();
+  test.strictSame(await api.collection.geo['1'].schemaCity.invoke('context'), {
+    context: 'context',
+    data: { name: 'string ' },
```
Comment (tshemsedinov) [inline test/interfaces.js:25]:
```suggestion
    data: { name: 'string' },
```

File: lib/interfaces.js (hunk 2 above)
Comment (georgolden) [reply lib/interfaces.js:103]: Here is a diff from hooks. For hooks script is already done earlier and hooks loaded within the existing script. For plugins we need to create Procedures for each plugin within the plugin script, not the file from api folder

File: package.json (hunk 3 above)
Comment (georgolden) [reply package.json:70]: This is only for testing, I referenced, that there must be cherry-pick only

**Review by tshemsedinov — APPROVED** [review]

**Comment by tshemsedinov** [comment]
Landed in https://github.com/metarhia/impress/commit/d31792cad01e2bf213b7fcf00bf57bbc2ae237e3

**Pull request closed without merging**